### PR TITLE
[server] Cleanup movedImageBuilder flag

### PR DESCRIPTION
**Pull request by WVerlaek**
## Description

Removes the `movedImageBuilder` ConfigCat flag, always building images in workspace clusters, as the flag has been enabled at 100% for over a week now.

Also updates dependency injection to bind `ImageBuilderClientProvider ` to a `WorkspaceClusterImagebuilderClientProvider`.

Tested that this still works for single-install clusters (i.e. a `full` install), as the `WorkspaceClusterImagebuilderClientProvider` will create a client for its own cluster (tested below).

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/9143

## How to test

Tested as follows:
* Create `-f full` [workspace-preview](https://github.com/gitpod-io/workspace-preview) cluster, and ran a successful image build.
* Create a `-f meta` (doesn't contain ws components) with a `-f workspace` workspace-preview cluster, and ran a successful image build.

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [x] /werft with-large-vm
- [ ] /werft with-integration-tests=webapp
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by kylos101** [comment]
@WVerlaek can you run one more test?

Create a -f full (does contain ws components) with a -f workspace workspace-preview cluster, asserting that the image build lands on the worksapce cluster.

I ask because :point_up: will mimic how production is setup "now".

**Comment by WVerlaek** [comment]
@kylos101 did a test to confirm: `server` picks a workspace cluster from the list of registered workspace clusters with server. In gitpod.io this would only be one of the workspace generation clusters (as `skipSelf` is enabled for the application cluster, excluding the application cluster from that list). 

In a workspace-preview `-f full` cluster, `skipSelf` isn't enabled, and the `-f full` cluster is included as one of the target clusters to run image builds (and workspaces). 

TLDR: image builds will run in one of the registered workspace clusters with the application cluster. This could be the application cluster itself as well, only if `skipSelf` isn't enabled (so not in gitpod.io).

File: components/server/src/container-module.ts [hunk 1]
```diff
@@ -178,9 +177,8 @@ export const productionContainerModule = new ContainerModule((bind, unbind, isBo
         const config = ctx.container.get<Config>(Config);
         return { address: config.imageBuilderAddr };
     });
-    bind(CachingImageBuilderClientProvider).toSelf().inSingletonScope();
-    bind(WorkspaceClusterImagebuilderClientProvider).toSelf().inSingletonScope(); // during the transition period, we have two kinds of image builder client providers
-    bind(ImageBuilderClientProvider).toService(CachingImageBuilderClientProvider);
+    bind(WorkspaceClusterImagebuilderClientProvider).toSelf().inSingletonScope();
+    bind(ImageBuilderClientProvider).toService(WorkspaceClusterImagebuilderClientProvider);
```
Comment (WVerlaek) [inline components/server/src/container-module.ts:181]: this changes the client type of any `@Inject(ImageBuilderClientProvider)` to the workspace cluster client provider, instead of the application cluster client provider (`CachingImageBuilderClientProvider`)

**Comment by WVerlaek** [comment]
The e2e tests keep failing with `no suitable user found: make sure there's at least one non-builtin user in the database (e.g. login)`, do I need to manually add a user?

**Comment by geropl** [comment]
Testing in the preview env...

**Review by geropl — APPROVED** [review]
Code LGTM, tested and work :heavy_check_mark:

**Comment by WVerlaek** [comment]
@geropl thanks for the review! Could you help me run the e2e tests? They keep failing on `"cannot access Gitpod server API: \"no suitable user found: make sure there's at least one non-builtin user in the database (e.g. login)\""`, not sure how to fix this?

**Comment by geropl** [comment]
/werft run

> Could you help me run the e2e tests?

Not sure, as I didn't do this in quite some time.

**Comment by geropl** [comment]
@WVerlaek I don't think we should wait for the integration tests. As sad as it is their are not used atm, so have low significance for this PR anyway.

**Comment by WVerlaek** [comment]
@geropl ok, I'll disable them in that case. Thanks
